samcli/lib/providers/sam_function_provider.py

```diff
@@ -235,12 +265,52 @@ def _convert_sam_function_resource(
             LOG.debug("Found Serverless function with name='%s' and ImageUri='%s'", name, imageuri)
 
         return SamFunctionProvider._build_function_configuration(
-            stack, name, codeuri, resource_properties, layers, inlinecode, imageuri, use_raw_codeuri
+            stack, function_id, name, codeuri, resource_properties, layers, inlinecode, imageuri, use_raw_codeuri
         )
 
+    @staticmethod
+    def _get_function_id(resource_properties: Dict, logical_id: str) -> str:
```

Comment:
I believe the metadata generated by CDK is not in the properties, it is on the resource's root level. Can you double check this please?

Reply:
You are right, in the template it is. We pull it into the properties in the function provider: https://github.com/aws/aws-sam-cli/blob/ec9f5d89971b823fa01071c96b7de90ea5e5303a/samcli/lib/providers/sam_function_provider.py#L133

Reply:
I see, thanks!